src/dmd/parse.d

```diff
@@ -1445,25 +1446,29 @@ final class Parser(AST) : Lexer
      */
     private StorageClass parseAttribute(ref AST.Expressions* udas)
     {
+        const Loc udastart = token.loc;
         nextToken();
         if (token.value == TOK.identifier)
         {
             // If we find a builtin attribute, we're done, return immediately.
             if (StorageClass stc = isBuiltinAtAttribute(token.ident))
                 return stc;
 
+            bool parens = true;
+
             // Allow identifier, template instantiation, or function call
             // for `@Argument` (single UDA) form.
             AST.Expression exp = parsePrimaryExp();
             if (token.value == TOK.leftParenthesis)
             {
                 const loc = token.loc;
                 exp = new AST.CallExp(loc, exp, parseArguments());
+                parens = false;
             }
 
             if (udas is null)
                 udas = new AST.Expressions();
-            udas.push(exp);
+            udas.push(new AST.UadItem(udastart, exp, parens));
```

Comment:
And why can't `exp` have its loc set to udastart?

Reply:
Because udas can be written in any of these ways: @exp, @(exp), @(exp1, exp2)
In the first 2 cases changing the loc of the exp would work but in the last it would be confusing to set all of them to the loc of  '@', especially when the expressions are on different lines.

Reply:
I also think that introducing a new AST type to represent this is overkill, and will not scale well.